cpp/include/raft/core/mdspan.hpp

```diff
@@ -222,4 +222,112 @@ RAFT_INLINE_FUNCTION auto unravel_index(Idx idx,
   }
 }
 
-}  // namespace raft
\ No newline at end of file
+template <class ElementType, class Extents, class Layout, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, Layout, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, layout_left, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, layout_right, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, layout_stride, Accessor> m)
+{
+  return m.is_exhaustive();
+}
+
+template <class ElementType, class Extents, class Layout, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, Layout, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, layout_left, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, layout_right, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, layout_stride, Accessor> m)
+{
+  return m.is_exhaustive();
+}
+
+template <class ElementType, class Extents, class Layout, class Accessor>
+constexpr bool is_col_major(mdspan<ElementType, Extents, Layout, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_col_major(mdspan<ElementType, Extents, layout_left, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_col_major(mdspan<ElementType, Extents, layout_right, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_col_major(mdspan<ElementType, Extents, layout_stride, Accessor> m)
+{
+  return m.is_exhaustive();
+}
+
+template <class ElementType, class IndexType>
+constexpr bool is_matrix_view(mdspan<ElementType, matrix_extent<IndexType>> /* m */)
```

Comment:
[required] There are a couple issues with the `is_{matrix,vector,scalar}_view` predicates.

1. This specialization assumes the default Layout (`layout_right`) and Accessor (`default_accessor`).  (Not specifying default template arguments means that the specialization is _only_ for the defaults.)
2. `is_matrix_view` really means `Extents::rank() == 2`.  `is_vector_view` means `Extents::rank() == 1`.  `is_scalar_view` means `Extents::rank() == 0`.  That's really all you need to check.

For example:

```c++
template <class ElementType, class IndexType, size_t ... Exts, class Layout, class Accessor>
constexpr bool is_matrix_view(mdspan<ElementType, extents<IndexType, Exts...>, Layout, Accessor> /* m */)
{
  return sizeof...(Exts) == 2; // 1 for is_vector_view, 0 for is_scalar_view
}
```